coconut/control/control.go

```diff
@@ -490,13 +490,89 @@ func QueryRoles(cxt context.Context, rpc *coconut.RpcClient, cmd *cobra.Command,
 
 // ListWorkflowTemplates lists the available workflow templates and the git repo on which they reside.
 func ListWorkflowTemplates(cxt context.Context, rpc *coconut.RpcClient, cmd *cobra.Command, args []string, o io.Writer) (err error) {
-	if len(args) != 0 {
-		err = errors.New(fmt.Sprintf("accepts no arg(s), received %d", len(args)))
-		return err
+	repoPattern := "*"
+	revisionPattern := "master"
+	allBranches := false
+	allTags := false
+
+	if len(args) == 0 {
+		repoPattern, err = cmd.Flags().GetString("repo")
+		if err != nil {
+			return
+		}
+		if len(repoPattern) == 0 {
+			err = errors.New("cannot query for an empty repo")
+			return
+		}
+
+		revisionPattern, err = cmd.Flags().GetString("revision")
+		if err != nil {
+			return
+		}
+
+		if len(revisionPattern) == 0 {
+			err = errors.New("cannot query for an empty revision")
+			return
+		}
+
+		allBranches, err = cmd.Flags().GetBool("all-branches")
+		if err != nil {
+			return
+		}
+
+		allTags, err = cmd.Flags().GetBool("all-tags")
+		if err != nil {
+			return
+		}
+
+		if allBranches || allTags {
+			if revisionPattern != "master" {
+				fmt.Fprintln(o, "Ignoring `--all-{branches,tags}` flags, as a valid revision has been specified")
+				allBranches = false
+				allTags = false
+			} else  {
+				revisionPattern = "*"
+			}
+		}
+
+	} else 	if len(args) == 1 { // If we have an argument, give priority over the flags
+		slicedArgument := strings.Split(args[0], "@")
```

Comment:
This might be safer with a regex just in case the user passes repo-patter@

Reply:
I agree that a regex would be the most elegant way to make a comprehensive but early check. The present example is likely low-impact though, as we already make sure that `len(slicedArgument)` is 1 or 2.